Base/QTCore/qSlicerExtensionsManagerModel.cxx

```diff
@@ -802,6 +811,138 @@ void qSlicerExtensionsManagerModelPrivate::saveExtensionDescription(
   qSlicerExtensionsManagerModel::writeExtensionDescriptionFile(extensionDescriptionFile, allExtensionMetadata);
 }
 
+
+// --------------------------------------------------------------------------
+void qSlicerExtensionsManagerModelPrivate::saveExtensionToHistorySettings(
+	const QString& extensionsHistorySettingsFile, const ExtensionMetadataType &extensionMetadata)
+{
+	this->addExtensionHistorySetting(extensionsHistorySettingsFile, extensionMetadata, "ExtensionsHistory/Revisions/" + this->SlicerRevision);
+}
+
+// --------------------------------------------------------------------------
+void qSlicerExtensionsManagerModelPrivate::scheduleExtensionHistorySettingRemoval(
+	const QString& extensionsHistorySettingsFile, const ExtensionMetadataType &extensionMetadata)
+{
+	this->addExtensionHistorySetting(extensionsHistorySettingsFile, extensionMetadata, "ExtensionsHistory/ScheduledForRemoval");
```

Comment:
Thank you for your feedback :) When the removal is performed, it removes entries for the current slicer revision (see removeScheduledExtensionHistorySettings - line 852):
```
QStringList& historyList = settings.value("ExtensionsHistory/Revisions/" + this->SlicerRevision).toStringList();
for (unsigned int i = 0; i < scheduledForRemovalList.length(); i++)
{
	historyList.removeOne(scheduledForRemovalList.at(i));
}
historyList.removeDuplicates();
settings.setValue("ExtensionsHistory/Revisions/" + this->SlicerRevision, historyList);
```
I agree that this is too implicit and might become a problem in a setup with multiple slicer versions running side-by-side (e.g. removal scheduled in version a but might affect version b). I'll add the revision to this setting.